api/v1alpha1/lvmcluster_types.go

```diff
@@ -143,6 +143,18 @@ type DeviceSelector struct {
 	ForceWipeDevicesAndDestroyAllData *bool `json:"forceWipeDevicesAndDestroyAllData,omitempty"`
 }
 
+type resolverFunc func(string) (string, error)
+
+type DevicePath string
```

Comment:
What happens if Resolve gets called more than once? Is there a reason this is not a sync.OnceFunc? AFAIU, this is gonna be the same behavior as the current code, but I thought we agreed that it makes sense to only resolve once per reconcile